### PR TITLE
fix possible MissingFormatArgumentException in ShouldHaveMessage

### LINKED ISSUE #1574: escapePercent (%) missing in org.assertj.core.internal.Failures#assertionErrorMessage
#### Summary
when using
        assertThatThrownBy(() -> restCal(XYZ))
                .isInstanceOf(MyServiceErrorException.class)
                .hasMessage(expectedMessage)

it throws MissingFormatArgumentException

#### Example

java.util.MissingFormatArgumentException: Format specifier '%3A'

	at java.util.Formatter.format(Formatter.java:2519)
	at java.util.Formatter.format(Formatter.java:2455)
	at java.lang.String.format(String.java:2940)
	at org.assertj.core.util.Strings.formatIfArgs(Strings.java:86)
	at org.assertj.core.error.MessageFormatter.format(MessageFormatter.java:62)
	at org.assertj.core.error.BasicErrorMessageFactory.create(BasicErrorMessageFactory.java:109)
	at org.assertj.core.internal.Failures.assertionErrorMessage(Failures.java:142)
	at org.assertj.core.internal.Failures.failure(Failures.java:132)
	at org.assertj.core.internal.Throwables.assertHasMessage(Throwables.java:87)
	at org.assertj.core.api.AbstractThrowableAssert.hasMessage(AbstractThrowableAssert.java:66)

### PR DESCRIPTION
**Pull request by epeee**
#### Check List:
* Fixes #1574
* Unit tests : YES
* Javadoc with a code example (API only) : NA

### CODE REVIEW

File: src/main/java/org/assertj/core/error/ShouldHaveMessage.java [hunk 1]
```diff
@@ -38,7 +40,7 @@ private ShouldHaveMessage(Throwable actual, String expectedMessage) {
           "  <%s>%n" +
           "%n" +
           "Throwable that failed the check:%n" +
-          "%n" + getStackTrace(actual), // to avoid AssertJ default String formatting
+          "%n" + Strings.escapePercent(getStackTrace(actual)), // to avoid AssertJ default String formatting
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldHaveMessage.java]: style: prefer static import

File: src/test/java/org/assertj/core/error/ShouldHaveMessage_create_Test.java [hunk 2]
```diff
@@ -40,4 +40,21 @@ public void should_create_error_message() {
                                        Throwables.getStackTrace(actual));
   }
 
+  @Test
+  void should_create_error_message_escaping_percent() {
+    Throwable actual = new RuntimeException("%3A");
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/error/ShouldHaveMessage_create_Test.java]: add // GIVEN above

**Comment by joel-costigliola** [comment]
Thanks @epeee, just minor comments on the PR, once addressed you can squash and merge it.

File: src/main/java/org/assertj/core/error/ShouldHaveMessage.java (hunk 1 above)
Comment (epeee) [reply src/main/java/org/assertj/core/error/ShouldHaveMessage.java]: :+1:, done

File: src/test/java/org/assertj/core/error/ShouldHaveMessage_create_Test.java (hunk 2 above)
Comment (epeee) [reply src/test/java/org/assertj/core/error/ShouldHaveMessage_create_Test.java]: thx, done

**Pull request merged**